jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/FrameFlusher.java

```diff
@@ -185,7 +184,14 @@ public void onClose(Throwable cause)
     {
         try (AutoLock l = lock.lock())
         {
-            closedCause = cause == null ? CLOSED_CHANNEL : cause;
+            closedCause = cause == null ? new ClosedChannelException()
```

Comment:
Can you include a TODO here indicating that this is a temporary solution until we can avoid using a throwable as an expensive boolean.